public/app/features/plugins/PluginPage.tsx

```diff
@@ -0,0 +1,420 @@
+      if (find(meta.includes, { type: 'dashboard' })) {
+        tabs.push({
+          text: 'Dashboards',
+          icon: 'gicon gicon-dashboard',
+          url: path + '?tab=' + TAB_ID_DASHBOARDS,
+          id: TAB_ID_DASHBOARDS,
+        });
+      }
+    }
+
+    if (!defaultTab) {
+      defaultTab = tabs[0].id; // the first tab
+    }
+
+    const node = {
+      text: meta.name,
+      img: meta.info.logos.large,
+      subTitle: meta.info.author.name,
+      breadcrumbs: [{ title: 'Plugins', url: '/plugins' }],
+      url: path,
+      children: this.setActiveTab(query.tab as string, tabs, defaultTab),
+    };
+
+    this.setState({
+      loading: false,
+      plugin,
+      defaultTab,
+      nav: {
+        node: node,
+        main: node,
+      },
+    });
+  }
+
+  setActiveTab(tabId: string, tabs: NavModelItem[], defaultTabId: string): NavModelItem[] {
+    let found = false;
+    const selected = tabId || defaultTabId;
+    const changed = tabs.map(tab => {
+      const active = !found && selected === tab.id;
+      if (active) {
+        found = true;
+      }
+      return { ...tab, active };
+    });
+    if (!found) {
+      changed[0].active = true;
+    }
+    return changed;
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const prevTab = prevProps.query.tab as string;
+    const tab = this.props.query.tab as string;
+    if (prevTab !== tab) {
+      const { nav, defaultTab } = this.state;
+      const node = {
+        ...nav.node,
+        children: this.setActiveTab(tab, nav.node.children, defaultTab),
+      };
+      this.setState({
+        nav: {
+          node: node,
+          main: node,
+        },
+      });
+    }
+  }
+
+  renderBody() {
+    const { query } = this.props;
+    const { plugin, nav } = this.state;
+
+    if (!plugin) {
+      return <div>Plugin not found.</div>;
+    }
+
+    const active = nav.main.children.find(tab => tab.active);
+    if (active) {
+      // Find the current config tab
+      if (plugin.configTabs) {
+        for (const tab of plugin.configTabs) {
+          if (tab.id === active.id) {
+            return <tab.body meta={plugin.meta} query={query} />;
+          }
+        }
+      }
+
+      // Apps have some special behavior
+      if (plugin.meta.type === PluginType.app) {
+        if (active.id === TAB_ID_DASHBOARDS) {
+          return <DashboardImporter plugin={plugin.meta} />;
+        }
+
+        if (active.id === TAB_ID_CONFIG_CTRL && plugin.angularConfigCtrl) {
+          return <AppConfigCtrlWrapper app={plugin as AppPlugin} />;
+        }
+      }
+    }
+
+    return <PluginHelp plugin={plugin.meta} type="help" />;
+  }
+
+  showUpdateInfo = () => {
+    alert('TODO, open modal');
+
+    const modalScope: any = {}; // this.$scope.$new(true);
+    modalScope.plugin = this.state.plugin.meta;
+
+    appEvents.emit('show-modal', {
+      src: 'public/app/features/plugins/partials/update_instructions.html',
+      scope: modalScope,
+    });
+  };
```

Comment:
@torkelo -- this is my one real blocker, everything else just needs testing from someone else running apps :)

Any ideas?  is there a way to make a new $scope?  or to `show-modal` with a react component?  `update_instructions` would be pretty easy to convert